saleor/order/views.py

```diff
@@ -26,8 +29,25 @@ def details(request, token):
                                    'user')
     order = get_object_or_404(orders, token=token)
     groups = order.groups.all()
-    return TemplateResponse(request, 'order/details.html',
-                            {'order': order, 'groups': groups})
+    notes = order.notes.filter(is_public=True)
+    if order.status == 'open':
+        note_form = NoteForm(request.POST or None)
+        if request.method == 'POST':
+            if note_form.is_valid():
+                order.notes.create(
+                    user=request.user,
+                    content=note_form.cleaned_data.get('note', ''))
+                order_url = build_absolute_uri(
+                    reverse('order:details', kwargs={'token': order.token}))
+                email = order.get_user_current_email()
+                send_note_confirmation.delay(email, order_url)
```

Comment:
I might be missing something, but is is the right place to send information about note added?
Shouldn't it occur when staff added public note from the dashboard?

Reply:
There's no need to do that. We should be notifying them when staff users add notes. We could consider notifying staff users about new notes added by customers, but at the moment we don't send any emails to staff, so that may be a completely separate issue.